pkg/dockerclient/types.go

```diff
@@ -29,6 +29,8 @@ const (
 	RestartPolicyNameOnFailure     RestartPolicyName = "on-failure"
 )
 
+const RunOptionsNoPort int = -1
```

Comment:
This is a bit unclear. What happens No port (-1) is used? is it a 1:1 port forward or does docker pick a random port? This needs a comment at least

Reply:
No, the docker container shell is a int->int map for ports to be exposed. in order not to break the code with many changes, I've added a constant to differentiate between explicitly mapping a source->target ports or ask docker to randomly provide a random source port

I've added a comment above, to explain the above

Reply:
ah it's explicit in https://github.com/nuclio/nuclio/pull/1766/files#diff-0cd3cd85e27a7182d9bd2b9a36d11efbR162
I see